specs/guaranteed-gas-market.md

```diff
@@ -0,0 +1,143 @@
+# Guaranteed Gas Fee Market
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Gas Stipend](#gas-stipend)
+- [Limiting Guaranteed Gas](#limiting-guaranteed-gas)
+- [1559 Fee Market](#1559-fee-market)
+- [Rationale for burning L1 Gas](#rationale-for-burning-l1-gas)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+[Deposited transaction](./glossary.md#deposited-transaction) are transactions on L2
+that are initiated on L1. The gas that they use on L2 is bought on L1 via a gas burn
+or a direct payment. We maintain a fee market and hard cap on the amount of gas provided
+to all deposits in a single L1 block.
+
+The gas provided to deposited transactions is sometimes called "guaranteed gas". The
+gas provided to deposited transactions is unique in the regard that it is not
+refundable. It cannot be refunded as it is sometimes paid for with a gas burn and
+there may not be any ETH left to refund.
+
+The **guaranteed gas** is composed of a gas stipend, and of any guaranteed gas the user
+would like to purchase (on L1) on top of that.
+
+## Gas Stipend
+
+Because there is some cost to submitting the transaction and updating the basefee,
+we provide transactions with a small amount of free gas.
+
+If the user requests more `guaranteedGas` than the `gasStipend`, that gas will
+be bought with L1 ETH via a gas burn or by buying it directly. If they request
+less gas than the stipend, they will not be charged.
+
+We provide XX,XXX amount of gas in the stipend. TODO - should this be price
+rather than gas based??
```

Comment:
To consider:
- we want to guarantee that this amount is always enough to register the tx on L2 for replayability purposes
- but at the same time, this means the guaranteed gas could be "cheaper" than guaranteed gas or even regular L2 gas, meaning it could be optimal to make a lot of deposits
     - however, taking advantage of this requires some kind of L2 setup where a long computation can be split over many smaller one, which is not usually how things work

Given that, I think gas-based is the way to go.

Reply:
Yea, I lean to gas based because it makes more sense. The problem is that it mucks up the gas market. Somewhere in the 50k-100k gas range feels right.